### ISSUE TITLE
How to pre-train from scratch?

### ISSUE THREAD

**Issue by roastduck** [allenai/longformer#73]
It seems that converting the model from RoBERTa limits the model's capability (short sequence length, no dilation). I want to reproduce the full model in the paper (specifically, with hyper-parameters in Table 12). It will be appreciated if someone can publish the training scripts.

**Comment by karan1508** [comment]
I could also use some help regarding this.

**Comment by ibeltagy** [comment]
Table 12 has hyperparameters for the character-lm experiments, while RoBERTa is for the pretraining/finetuning setting. These are two different models with two different configurations and two different datasets.

- If you are referring to the pertaining/finetuning setting:

1. To train from scratch, you can do this by instantiating a randomly initialized `LongformerModel`, but you will need a huge amount of training data and compute resources. 

2. 
> short sequence length
If you want a pretraining/finetuning model (similar to the released pretrained model) but for longer sequences, you can still start from RoBERTa and build your own model following the instructions in this [notebook](https://colab.research.google.com/github/allenai/longformer/blob/master/scripts/convert_model_to_long.ipynb). I tested it and with gradient checkpointing, fp16, 512 window size, 48GB gpu, you can reach a sequence length of 64K. This is something that we are considering as well. 

3.
> no dilation
To add dilation to this model, you will have to use the `tvm` implementation (`sliding_chunks` and the HF code don't support dilation). You don't need to start from scratch to train this model. We briefly tried this configuration but it didn't outperform the no-dilation case, so we didn't investigate it further. It is possible that training the model for longer and on more training, and maybe on a different objective function can make it work. 


- If you are referring to the character-lm experiments, we didn't release implementation details for this model. You can follow the model description from the paper, but the implementation of relative position embedding + our sliding window attention is a little tricky. We are hoping to release this code in the near future.

**Comment by ibeltagy** [comment]
Closing. Please feel free to reopen if needed.

**Comment by amirubin87** [comment]
Thanks for that!
Quick comment - in the notebook you reference table 6 in the [longformer paper](https://arxiv.org/pdf/2004.05150.pdf), seems like it should be table 5:
![image](https://user-images.githubusercontent.com/8530559/171448082-526dfe30-3a92-47f3-bbe6-9e8021bc3ea8.png)
